Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -691,6 +694,134 @@ public class CameraManager {
+     - Parameter delay: The number of seconds to wait before starting the animations. Specify 0 to begin the animations immediately.
+     - Parameter options: The options to apply to the animations. You can specify most options, but transition-related options and options related to the animation direction are ignored.
+     - Parameter animationOwner: Property that conforms to `AnimationOwnerProtocol` to represent who owns that animation.
+     - Parameter animations: The block containing the animations. This block has no return value and takes no parameters. Use this block to modify any animatable view properties. When you start the animations, those properties are animated from their current values to the new values using the specified animation parameters.
+     - Parameter completion: Completion block to be passed through for when an animation is stopped
+     - Returns `CameraAnimator`: A class that represents an animator with the provided configuration.
+     */
+    public func runningCameraAnimator(duration: TimeInterval,
+                                      delay: TimeInterval,
+                                      options: UIView.AnimationOptions = [],
+                                      animationOwner: AnimationOwnerProtocol = AnimationOwner.unspecified,
+                                      animations: @escaping () -> Void,
+                                      completion: AnimationCompletion? = nil) -> CameraAnimator {
+        let runningAnimator = UIViewPropertyAnimator.runningPropertyAnimator(withDuration: duration,
+                                                                             delay: delay,
+                                                                             options: options,
+                                                                             animations: animations,
+                                                                             completion: nil)
+
+        let cameraAnimator = CameraAnimator(delegate: self, propertyAnimator: runningAnimator, owner: animationOwner)
+
+        runningAnimator.addCompletion({ [weak self] animatingPosition in
+            guard let completion = completion,
+                  let self = self
+            else { return }
+
+            self.schedulePendingCompletion(forAnimator: cameraAnimator, completion: completion, animatingPosition: animatingPosition)
+        })
+        cameraAnimators.add(cameraAnimator)
+        return cameraAnimator
+    }
+
+    // MARK: Delegate Functions
+    func schedulePendingCompletion(forAnimator animator: CameraAnimator, completion: @escaping AnimationCompletion, animatingPosition: UIViewAnimatingPosition) {
+        guard let mapView = mapView else { return }
+        mapView.pendingAnimatorCompletionBlocks.append((completion, animatingPosition))
+        cameraAnimators.remove(animator)
+    }
+
+    func animatorIsFinished(forAnimator animator: CameraAnimator) {
+        cameraAnimators.remove(animator)
```

Comment:
Hmm, do we need `animatorIsFInished` at all? Seems like `schedulePendingCompletion` takes care of it all?

Reply:
What about the case that StopAnimation is manually called and that would trigger these changes to occur. Which is when `animatorIsFinished` gets called

Reply:
Hmm but doesn't `schedulePendingCompletion` get called automatically in that case?

Reply:
Ah wait -- an animation might not have a completion. Never mind, this is necessary.

Reply:
Let's just change it so as to check if the animator exists in the hashTable.. If it does, remove it. Else, do nothing.